README.md
# ELS - Enterprise Linux/EuroLinux Sources - perl-Exception-Class
 
## Usage:
  Checkout branch or tag. Use `make sources` to recreate  SOURCE0 archive.


src/Changes
1.37    2013-02-24

- I now recommend you use Throwable instead of this module. It has a nicer,
  more modern interface.

- Fixed various bugs and confusion in the docs.


1.36    2012-11-17

- Fixed some stupidity in the tests that appears to have been highlighted by
  recent changes to Devel::StackTrace. Reported by Dan Horne. RT #81245.


1.35    2012-09-17

- Require Class::Data::Inheritable 0.02+.


1.34    2012-09-16

- 1.33 did not declare any prereqs. Oops. Reported by Roham McGovern. RT
  #79677.


1.33    2012-09-16

- Fixed warning from basic.t on 5.17.x. RT #79121.


1.32    2010-06-28

- Removed Test::Most dependency, which was causing a circular dependency
  chain. Reported by Burak Gursoy. RT #58889.


1.31    2010-06-26

- Made it possible to use Exception::Class::Base without loading
  Exception::Class. Reported by Alex Peters. RT #57269.


1.30    2010-03-20

- Added the ability to create light weight exceptions, which don't record a
  stack trace or any other context info (time, pid, etc.). This is done by
  setting $class->NoContextInfo to a true value for the exception class. Based
  on a patch by Graham Barr. RT #54826.


1.29    2009-05-08

- Make sure that there is only one line that the toolchain can pick up
  when looking for this module's $VERSION.


1.28    2009-05-06

* Removed Exception::Class::Base->do_trace and ->NoObjectRefs, both of
  which have been undocumented for many years.

- Moved Exception::Class::Base to its own file, so it doesn't
  overwrite the $VERSION in Exception::Class (and for general
  sanity). Reported by Kirk Baucom.


1.27    2009-05-04

* The error message for an exception no longer defaults to $!. This
  was a bad idea, and has bitten some people. Addresses RT #43600.

* I'm now only supporting Perl 5.8.1+. This module broke on 5.6.2 (and
  5.8.0) at some point and I don't really care enough to fix
  it. Non-intrusive patches to make it work on 5.8.0 or less (again)
  will be accepted.


1.26    2008-10-25

- No code changes, just bumped the Devel::StackTrace dependency to
  1.20.

  The changes in 1.25 exposed a bug in Devel::StackTrace which has now
  been fixed. This caused HTML::Mason test failures, among other
  things. Reported by Andreas Koenig.


1.25    2008-10-21

- The Exception::Class::Base constructor no longer looks at the
  associated Devel::StackTrace object's frames. Avoiding looking at
  the frames should make constructing exception objects much
  faster. Instead the information based on the stack trace frames -
  package(), file(), and line() - is only retrieved when it is
  requested. Patch by Ruslan Zakirov. RT #40221.


1.24    2008-03-30

- Added a MaxArgLength class parameter to go along with the new
  max_arg_length constructor param for Devel::StackTrace
  objects. Patch by Ian Burrell.


1.23    2006-01-14

- Fix code bugs in the SYNOPSIS.

- Exception::Class->caught() now returns $@ if not given an argument.


1.22    2005-09-30

- Added Exception::Class::Base->caught() so you can write
  "My::Error->caught()".  Apparently this is what Damian documented in
  Perl Best Practices.  Thanks to JD Hedden for pointing this out.

- Require Devel::StackTrace 1.12 to fix a test failure on Win32.
  Reported by Garrett Goebel.


1.21    2005-04-17

- Added Exception::Class->caught() as syntactic sugar for catching
  exceptions in a "safe" manner.  See the docs for details.  Suggested
  by Damian Conway.


1.20    2005-01-01

- Moved the Classes() method to Exception::Class, now as a function,
  which is where it really belongs.  Also corrected the docs for this
  function, as they incorrectly implied that it would return all
  subclasses, instead of just subclasses which were created when
  loading Exception::Class.

- Added a Build.PL file, and releases are now signed with
  Module::Signature.

- Added ignore_class and ignore_package constructor parameters.  Patch
  by Daisuke Maki.


1.19    2004-03-10

- Identical to 1.18 except that it requires Devel::StackTrace 1.10,
  which fixes a failure in this package's test when run with Perl
  5.6.1.  Reported by Jesse Erlbaum.


1.18    2004-02-21

- Added RespectOverload class method, which can be used to control the
  respect_overload parameter for Devel::StackTrace objects.


1.17    2004-02-20

- Document that new() is always called to create an
  Exception::Class::Base object.  Suggested by Steve Hay.

- Fix a test that failed with Devel::StackTrace 1.05.  Basically, the
  test expected output from Devel::StackTrace that reflected a bug
  that was fixed in 1.05.  Reported by Jesse Vincent.


1.16    2003-09-25

- David Wheeler's patch exposed a bug in Devel::StackTrace which could
  cause exceptions to not have any value set for package, file, or
  line.  This is fixed by depending on Devel::StackTrace 1.04.  This
  release is otherwise identical to 1.15.  Reported by Steve Hay.


1.15    2003-09-17

- Exceptions thrown from an alias subroutine always had
  "Exception::Class" as their package name.  Patch by David Wheeler.


1.14    2003-07-04

- ** BACKWARDS INCOMPATIBILITY **
  NoRefs is now true by default.  This means that stacktraces will
  _not_ store references unless you explicitly set NoRefs to a false
  value.  This is done because it's too easy to leak memory when
  references are saved in stack traces.

- Fixed a very odd bug where Exception::Class could mistakenly think
  that an exception class had been created when it really hadn't.
  This was only exposed with bleadperl (5.8.1-to-be), and is unlikely
  to affect anyone using an existing Perl version.


1.13    2003-06-21

- Documented the Exception::Class::Base->Fields method, which can be
  called on subclasses to determine what extra fields they have
  defined.  Suggested by Chris Winters.


1.12    2003-04-03

- Added the Exception::Class::Base->Classes method.  Patch by David
  Wheeler.


1.11    2003-02-20

- A description containing a single quote or backslash would cause a

src/Makefile.PL

use strict;
use warnings;

use 5.008001;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "A module that allows you to declare real exception classes in Perl",
  "AUTHOR" => "Dave Rolsky <autarch\@urth.org>",
  "BUILD_REQUIRES" => {
    "File::Spec" => 0,
    "Test::More" => "0.88",
    "vars" => 0
  },
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Exception-Class",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Exception::Class",
  "PREREQ_PM" => {
    "Class::Data::Inheritable" => "0.02",
    "Devel::StackTrace" => "1.20",
    "Scalar::Util" => 0,
    "base" => 0,
    "overload" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "VERSION" => "1.37",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



